app/controllers/merge_controller.rb

```diff
@@ -90,19 +93,24 @@ def parse_rendundant_review_params
     set_merge_request
   end
 
-  # ^-- end merge GET param parsers
+  # ^-- end GET param parsers
 
-  def parse_merge_post_params
-    admins_only
+  def parse_post_params
     case @merge_mode
     when Modes::EXECUTE
+      admins_only
```

Comment:
fun rails terseness challenge: ` admins_only ` is used 5 times in this controller...is there a shorter way? :)

Reply:
hmm... are you suggesting a before action hook? i considered that but rejected it because the checks need to come inside of conditional blocks. (so `admin_only` would not always apply to one action or another).

is there another way i'm not seeing? a guard `unless` clause at the top of each `parse` helper? but that seems redundant with the case statement (by virtue of being two forms of conditional branches.)

we could extract each mode's logic into its own action and route to allow for 1-to-1 matching of permissions with route, but, as you recall, that was rejected at the outset as "too heavy handed."

Reply:
but, yes: i too do not like the repetetive invocations of permissions check inside each params parser. maybe a check_permissions before hook that gets called on each action and then groups the modes into an array of options that correspond to each permission? imma give that a try. but first i need to get my testing DB unborked. (see semaphor comment.)